public/locations-list-map.js

```diff
@@ -371,19 +598,25 @@ function showMarkers(data, filters, showNearest) {
   let $mapStats = $('#map-stats');
   updateStats($mapStats, markers.length);
 
+  // TODO: See notes above; this function should NOT be responsible for positioning the map.
+
   if (showNearest) {
     centerMapToNearestMarkers(map, markers, bounds);
   } else {
     centerMapToBounds(map, bounds, 9)
   }
 }
+window.showMarkers = showMarkers; // Exposed for debug/testing.
+
+
+
+// TODO (patricknelson): I propose completely scrapping this function in favor of centerMapToMarkersNearCoords() for simplicity.
```

Comment:
Different concerns. This method is to set the view to show all markers that pass the current filters.

Reply:
Conceptually, what we could do (as done in this version of the PR) would be to simply modify `markers` variable, which was moved to top level scope of the module. That's why we can still consistently shift between using "use my location", use the user-entered location in the search field or use the state-filtered markers and have it still work consistently. 

Basically a lot of intricate inter-dependencies can be resolved by maintaining state at the module level and passing less parameters to each function and just relying on that state.